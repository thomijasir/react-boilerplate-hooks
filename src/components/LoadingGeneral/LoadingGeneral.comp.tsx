import React, { FC } from 'react';
import LOADING_SPINNER from '../../assets/images/react.svg';
import './LoadingGeneral.scss';

export type ILoadingGeneralProps = {
  isLoading: boolean;
  text?: string;
};

export const loadingGeneralDefaultProps = {
  isLoading: false,
  text: 'Loading..',
};

export const loadingGeneralNamespace = 'LoadingGeneral';

const LoadingGeneral: FC<ILoadingGeneralProps> = ({ isLoading, text }) => (
  <div className={`loading-mask ${isLoading ? 'on' : ''}`}>
    <div className="spinner">
      <img className="rotate" src={LOADING_SPINNER} alt="loading" />
    </div>
    <div className="loading-text">{text}</div>
  </div>
);

LoadingGeneral.displayName = loadingGeneralNamespace;
LoadingGeneral.defaultProps = loadingGeneralDefaultProps;
export default React.memo(LoadingGeneral);
