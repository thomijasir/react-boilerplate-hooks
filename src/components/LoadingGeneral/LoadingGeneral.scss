@import '../../assets/style/colors';
@import '../../assets//style/var';
.loading-mask {
  position: fixed;
  z-index: -1;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  background-color: $white;
  display: flex;
  flex-direction: column;
  opacity: 0;
  transition-duration: 0.2s;
  text-align: center;
  justify-content: center;
  &.on {
    opacity: 1;
    z-index: 9999;
  }
}

.loading-text {
  margin-top: -18px; 
}

.rotate {
  width: 10rem;
  height: 10rem;
  animation: rotation 2s infinite linear;
}

@keyframes rotation {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(359deg);
  }
}
