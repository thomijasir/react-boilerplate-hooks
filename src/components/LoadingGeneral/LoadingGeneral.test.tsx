import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import LoadingGeneral from './LoadingGeneral.comp';

afterEach(() => {
  jest.clearAllMocks();
});

describe('Loading General Component Snap Test', () => {
  describe('render()', () => {
    test('renders the component', () => {
      const compProps = {
        isLoading: false,
      };
      const wrapper = shallow(<LoadingGeneral {...compProps} />);
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });
});
