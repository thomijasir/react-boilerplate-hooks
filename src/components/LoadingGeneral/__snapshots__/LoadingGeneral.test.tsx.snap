// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading General Component Snap Test render() renders the component 1`] = `
<div
  className="loading-mask "
>
  <div
    className="spinner"
  >
    <img
      alt="loading"
      className="rotate"
      src="react.svg"
    />
  </div>
  <div
    className="loading-text"
  >
    Loading..
  </div>
</div>
`;
